rcl/src/rcl/logging_rosout.c

```diff
@@ -243,20 +246,31 @@ rcl_ret_t rcl_logging_rosout_init_publisher_for_node(rcl_node_t * node)
   // Verify input and make sure it's not already initialized
   RCL_CHECK_ARGUMENT_FOR_NULL(node, RCL_RET_NODE_INVALID);
   logger_name = rcl_node_get_logger_name(node);
+  const char * node_name = rcl_node_get_name(node);
```

Comment:
I think this should be moved into the `if (rcutils_hash_map_key_exists)` section, since we don't need it elsewhere.  We should also double-check whether the `node_name` is NULL before using it, as that is an (unlikely) situation that can happen.